src/fmm.c

```diff
@@ -1390,7 +1389,7 @@ void fmm_release(void *address)
 			vm_remove_object(&cpuvm_aperture, object);
 		pthread_mutex_unlock(&cpuvm_aperture.fmm_mutex);
 		/* Free the memory from the system */
-		free(address);
+		munmap(address, MemorySizeInBytes);
```

Comment:
It would be safer to use the size stored in the vm_object_t instead of relying on the value specified by the caller. It would also simplify the patch because you wouldn't need to change all the callers of fmm_release.